addons/web_editor/static/src/js/wysiwyg/wysiwyg.js

```diff
@@ -236,7 +236,7 @@ const Wysiwyg = Widget.extend({
         this._updateEditorUI();
 
         this.$root.on('click', (ev) => {
-            const $target = $(ev.target).closest('a');
+            const $target = $(ev.target).closest(`a:not(:has(${mediaSelector}))`);
```

Comment:
Indeed, the conflict was not with the popup but with the link tools in the toolbar (even though they are not displayed).
I adapted the behavior so that the toolbar edit & remove link buttons relate to the image link instead of the text link in that situation.